packages/suite/src/components/suite/bluetooth/BluetoothConnect.tsx

```diff
@@ -0,0 +1,199 @@
+};
+
+const selectAllDevices = prepareSelectAllDevices<DesktopBluetoothDevice>();
+
+export const BluetoothConnect = ({ uiMode }: BluetoothConnectProps) => {
+    const dispatch = useDispatch();
+
+    const [waitingForFirstUpdate, setWaitingForFirstUpdate] = useState<boolean>(true);
+    const [selectedDeviceId, setSelectedDeviceId] = useState<string | null>(null);
+    const [scannerTimerId, setScannerTimerId] = useState<TimerId | null>(null);
+
+    const bluetoothAdapterStatus = useSelector(selectAdapterStatus);
+    const allDevices = useSelector(selectAllDevices);
+    const knownDevices = useSelector(selectKnownDevices);
+    const nearbyDevices = useSelector(selectNearbyDevices);
+
+    const lasUpdatedBoundaryTimestamp =
+        Date.now() / 1000 - UNPAIRED_DEVICES_LAST_UPDATED_LIMIT_SECONDS;
+
+    const devices = allDevices.filter(it => {
+        const isDeviceUnresponsiveForTooLong =
+            it.lastUpdatedTimestamp < lasUpdatedBoundaryTimestamp;
+
+        if (isDeviceUnresponsiveForTooLong) {
+            // If the device is connected or paired (it may have been paired in the OS system directly)
+            // => do not filter it based isDeviceUnresponsiveForTooLong
+
+            return (
+                it.connected ||
+                it.paired ||
+                knownDevices.find(knownDevice => knownDevice.id === it.id) !== undefined
+            );
+        }
+
+        return true;
+    });
+
+    const selectedDevice =
+        selectedDeviceId !== null
+            ? devices.find(device => device.id === selectedDeviceId)
+            : undefined;
+
+    useEffect(() => {
+        dispatch(bluetoothStartScanningThunk());
+
+        return () => {
+            dispatch(bluetoothStopScanningThunk());
+        };
+    }, [dispatch]);
+
+    const clearScamTimer = useCallback(() => {
+        if (scannerTimerId !== null) {
+            clearTimeout(scannerTimerId);
+        }
+    }, [scannerTimerId]);
+
+    useEffect(() => {
+        // Intentionally no `clearScamTimer`, this is first run and if we use this we would create infinite re-render
+        const timerId = setTimeout(() => {
+            dispatch(bluetoothActions.scanStatusAction({ status: 'idle' }));
+        }, SCAN_TIMEOUT);
+
+        setScannerTimerId(timerId);
+    }, [dispatch]);
+
+    useEffect(() => {
+        if (devices.length > 0) {
+            clearScamTimer();
+            dispatch(bluetoothActions.scanStatusAction({ status: 'idle' }));
+        }
+    }, [devices, dispatch, clearScamTimer]);
+
+    // When we have some knownDevices, it may happen that we get empty nearbyDevices first and then,
+    // in split second we get the update and there are some nearbyDevices present.
+    // Device in knownDevices but not in the nearbyDevices is recognized as to-be-deleted,
+    // and ui flickers. This small loading is here to hide that.
+    useEffect(() => {
+        setTimeout(() => setWaitingForFirstUpdate(false), 100);
```

Comment:
Nit. Yeah I realize chance of unmounting component so soon is negligible, but still a good practice IMO
```suggestion
        const timeoutHandle = setTimeout(() => setWaitingForFirstUpdate(false), 100);
        
        return () => clearTimeout(timeoutHandle); 
```